### PR TITLE
Dev branch merge into Main

### PR DESCRIPTION
**Pull request by IdrisDose**
**Describe Pull request**
This PR implements some new functionality as well as some fixes.

**Features**
- DrawText is now implemented into Core meaning you can display text on either top, left or right. See Issue #527
- Dynamic Shared Tables - Added the ability to dynamically add gangs, jobs and items to the core at resource startup without having to create a new shared entry. See issue #530 

**Fixes**
- Added missing vehicle properties.

### CODE REVIEW

File: client/drawtext.lua [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        action = 'CHANGE_TEXT',
+        data = {
+            text = text,
+            position = position
+        }
+    })
+end
+
+local function keyPressed()
+    CreateThread(function() -- Not sure if a thread is needed but why not eh?
```
Comment (deandum) [inline client/drawtext.lua:32]: Doesn't this start a thread for each key pressed? I don't think this is needed. I tested it locally and it works fine without it.

Comment (BerkieBb) [reply client/drawtext.lua:32]: That's not how that works

Comment (BerkieBb) [reply client/drawtext.lua:32]: But yeah it's unnecessary

**Comment by JonasDev99** [comment]
On line 600 there is a += which afaik there is no += in lua.

```
function QBCore.Player.GetTotalWeight(items)
    local weight = 0
    if not items then return 0 end
    for _, item in pairs(items) do
        weight += item.weight * item.amount
    end
    return tonumber(weight)
end
```

**Comment by BerkieBb** [comment]
> On line 600 there is a += which afaik there is no += in lua.
> 
> ```
> function QBCore.Player.GetTotalWeight(items)
>     local weight = 0
>     if not items then return 0 end
>     for _, item in pairs(items) do
>         weight += item.weight * item.amount
>     end
>     return tonumber(weight)
> end
> ```

There is in cfxlua
https://github.com/citizenfx/lua/tree/luaglm-dev/cfx
Cfxlua has a lot which a lot of people don't knoe about

**Comment by JonasDev99** [comment]
Oh, didn't know, nice 👍

**Pull request merged**